bgls/simulator_test.py

```diff
@@ -278,3 +280,60 @@ def test_remains_clifford():
     sim_results_stab = sim_act_on_stab.run(clifford_circuit, repetitions=100)
 
     assert sim_results == sim_results_stab
+
+
+def test_mps_results_match_state_vec():
+    """Test sampled bitstrings are same when using a matrix product state
+    simulator and a state vector simulator.
+    """
+    try:
+        import quimb.tensor as qtn
+        import cirq.contrib.quimb.mps_simulator
+    except ImportError:
+        return "quimb not installed, mps simulation not possible"
+
+    def cirq_mps_bitstring_probability(
+        mps: cirq.contrib.quimb.MPSState, bitstring: str
+    ) -> float:
+        """
+        Returns the probability of measuring the `bitstring` (|z⟩) in the
+        'cirq.contrib.quimb.MPSState' mps.
+        Args:
+            mps: Matrix Product State as a 'cirq.contrib.quimb.MPSState'.
+            bitstring: Bitstring |z⟩ as a binary string.
+        """
+        M_subset = []
+        for i, Ai in enumerate(mps.M):
+            qubit_index = mps.i_str(i)
+            # selecting the component with matching bitstring:
+            A_subset = Ai.isel({qubit_index: int(bitstring[i])})
+            M_subset.append(A_subset)
+
+        tensor_network = qtn.TensorNetwork(M_subset)
+        state_vector = tensor_network.contract(inplace=False)
```

Comment:
This should be an amplitude after contracting, no? Why call it `state_vector`?

Reply:
Ah apologies yes - the contraction yields a scalar rather than a Tensor, but I will update this name to match being an amplitude.